BaragonAgentService/src/main/java/com/hubspot/baragon/agent/lbs/FilesystemConfigHelper.java

```diff
@@ -63,13 +63,22 @@ public void checkAndReload() throws InvalidConfigException, LbAdapterExecuteExce
     final BaragonService service = context.getService();
     final boolean previousConfigsExist = configsExist(service);
     Collection<BaragonConfigFile> newConfigs = configGenerator.generateConfigsForProject(context);
-    if (previousConfigsExist && newConfigs.equals(readConfigs(service))) {
+    if (previousConfigsExist && configsMatch(newConfigs, readConfigs(service))) {
       return Optional.absent();
     } else {
       return Optional.of(newConfigs);
     }
   }
 
+  public boolean configsMatch(Collection<BaragonConfigFile> newConfigs, Collection<BaragonConfigFile> currentConfigs) {
+    for (BaragonConfigFile file : newConfigs) {
+      if (!currentConfigs.contains(file)) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
This is checking that `newConfigs` is a subset of `currentConfigs`, is it possible that its a strict superset?